run_operations.py

```diff
@@ -1,394 +1,293 @@
-
-	""" initiate directories for run_operations.py """
-	operations_directory = os.getenv('OPERATIONS')
-	os.makedirs(operations_directory, exist_ok=True)
-	os.makedirs(operations_directory + "/TEMPLATES/", exist_ok=True)
-	os.makedirs(operations_directory + "/ERRORS/", exist_ok=True)
-	open(os.getenv('OPERATIONS') + '/stored_date.date', 'a').close()  # create empty file
-
-	os.makedirs(operations_directory + "/LOGS/", exist_ok=True)
-	open(operations_directory + "/LOGS/" + '/generate_templates.log', 'a').close()  # create empty file
-           
+    ssaraopt_cmd = 'ssara_federated_query.py {} --print'.format(ssaraopt_string)
 
-if __name__ == "__main__":
+    ssara_output = subprocess.check_output(ssaraopt_cmd, shell=True)
+
+    newest_data = ssara_output.decode('utf-8').split("\n")[-2]
+    date = datetime.strptime(newest_data.split(",")[3], DATE_FORMAT)
+
+    return date
+
+def get_last_downloaded_date(dset):
+
+    dataset_line = None
+    with open(STORED_DATE_FILE) as date_file:
+        for line in date_file.readlines():
+            if dset in line:
+                dataset_line = line
+                break
+
+    if dataset_line is not None:
+        last_date = dataset_line.split(": ")[1].strip("\n")
+    else:
+        last_date = datetime.strftime(datetime(1970, 1, 1, 0, 0, 0), DATE_FORMAT)
+    
+    return datetime.strptime(last_date, DATE_FORMAT)
+
+def run_process_rsmas(inps, template_file, dataset):
+
+    process_rsmas_options = []
+
+    if inps.startssara:
+        process_rsmas_options.append('--startssara')
+    if inps.stopssara:
+        process_rsmas_options.append('--stopssara')
+    if inps.startprocess:
+        process_rsmas_options.append('--startprocess')
+    if inps.stopprocess:
+        process_rsmas_options.append('--stopprocess')
+    if inps.startpysar:
+        process_rsmas_options.append('--startpysar')
+    if inps.stoppysarload:
+        process_rsmas_options.append('--stoppysarload')
+    if inps.stoppysar:
+        process_rsmas_options.append('--stoppysar')
+    if inps.startinsarmaps:
+        process_rsmas_options.append('--startinsarmaps')
+
+    if len(process_rsmas_options) == 0:
+        process_rsmas_options.append('--insarmaps')
+
+    process_rsmas_options = ' '.join(process_rsmas_options)
+
+    process_rsmas_cmd = "process_rsmas.py {} {} --submit --update".format(template_file, process_rsmas_options)
+
+    process_rsmas = subprocess.check_output(process_rsmas_cmd, shell=True).decode('utf-8')
+    
+    job_number = process_rsmas.split('\n')[0]
+
+    stdout_file = "{}/{}/process_rsmas_{}.o".format(os.getenv('SCRATCHDIR'), dataset, job_number)
+    stderr_file = "{}/{}/process_rsmas_{}.e".format(os.getenv('SCRATCHDIR'), dataset, job_number)
+
+    logger_run_operations.log(loglevel.INFO, "Job Number: {}".format(job_number))
+    logger_run_operations.log(loglevel.INFO, "Output Files: {}, {}".format(stdout_file, stderr_file))
+
+    return [stdout_file, stderr_file], job_number
```

Comment:
Make return `[stdout_file, stderr_file]` into a tuple

Reply:
@2gotgrossman Decided I'm not going to do this. It doesn't increase readability or functionality in any way I can see. I think its more obvious what happening if returned as list, since its concatenated into a list.